test/unit/deletion_test.rb

```diff
@@ -3,42 +3,52 @@
 class DeletionTest < ActiveSupport::TestCase
   self.use_transactional_fixtures = false # Disabled to test after_commit
 
+  should belong_to :user
+
   setup do
     @user = create(:user)
     Pusher.new(@user, gem_file).process
     @version = Version.last
   end
 
-  test "must be indexed" do
+  should "must be indexed" do
     @version.indexed = false
     assert Deletion.new(version: @version, user: @user).invalid?,
       "Deletion should only work on indexed gems"
   end
 
-  test "unindexes" do
-    delete_gem
-    assert !@version.indexed?
-  end
+  context "with deleted gem" do
+    setup do
+      delete_gem
+    end
 
-  test "be considered deleted" do
-    delete_gem
-    assert Version.yanked.include?(@version)
-  end
+    should "unindexes" do
+      assert !@version.indexed?
```

Comment:
@nateberkopec @arthurnn I updated as refute since both of you recommend it. Thanks for review, I guess this PR is ready to merge now